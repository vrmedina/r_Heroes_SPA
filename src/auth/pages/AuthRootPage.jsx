import { Outlet } from "react-router-dom";

export const AuthRootPage = () => {
  return (
    <>
      <Outlet />
    </>
  );
};
