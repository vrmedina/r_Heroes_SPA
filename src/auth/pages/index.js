export * from "./LoginPage";
export * from "./AuthRootPage";