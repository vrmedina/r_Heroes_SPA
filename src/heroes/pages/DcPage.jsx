import { HeroList } from "../components";

export const DcPage = () => {
  return (
    <>
      <h1 className="mt-3">DC Comics</h1>
      <hr />
      <HeroList publisher="DC Comics" />
    </>
  );
};
