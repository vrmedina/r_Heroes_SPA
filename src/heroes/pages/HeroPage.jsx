import { Navigate, useParams } from "react-router-dom";
import { getHeroById } from "../helpers";
import { HeroDetail } from "../components/HeroDetail";
import { useMemo } from "react";

export const HeroPage = () => {
  const { id } = useParams();
  const hero = useMemo(() => getHeroById(id), [id]);

  if (!hero) {
    return <Navigate to={"heroes/marvel"} />;
  }
  return (
    <div className="row mt-3">
      <HeroDetail key={hero.id} {...hero} />
    </div>
  );
};
