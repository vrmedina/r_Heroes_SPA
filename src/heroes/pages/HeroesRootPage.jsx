import { Outlet } from "react-router-dom";
import { Navbar } from "../../ui";

export const HeroesRootPage = () => {
  return (
    <>
      <Navbar />
      <Outlet />
    </>
  );
};
