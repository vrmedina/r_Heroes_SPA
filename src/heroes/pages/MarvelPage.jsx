import { HeroList } from "../components";

export const MarvelPage = () => {
  return (
    <>
      <h1>Marvel Comics</h1>
      <hr />
      <div className="container">
        <HeroList publisher="Marvel Comics" />
      </div>
    </>
  );
};
