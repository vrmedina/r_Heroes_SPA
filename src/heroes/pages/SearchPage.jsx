import { useLocation, useNavigate } from "react-router-dom";
import { useForm } from "../../hooks/useForm";
import queryString from "query-string";
import { getHeroesByName } from "../helpers/getHeroesByName";
import { HeroCard } from "../components/HeroCard";

export const SearchPage = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const { q = "" } = queryString.parse(location.search);
  const heroes = getHeroesByName(q);

  const showSearch = q.length === 0;
  const showError = q.length > 0 && heroes.length === 0;

  const { searchText, onInputChange } = useForm({
    searchText: q,
  });

  const onSearchSubmit = (evt) => {
    evt.preventDefault();
    navigate(`?q=${searchText}`);
  };

  return (
    <div className="container container-fluid">
      <h1 className="mt-3">Superhero Search</h1>
      <hr />
      <div className="card my-3" style={{ maxWidth: "1000px" }}>
        <div className="row g-0">
          <div className="col-xs-12 col-sm-5 col-md-5 col-lg-4">
            <div className="card-body">
              <h5 className="card-title">Search</h5>
              <hr />
              <form onSubmit={onSearchSubmit}>
                <input
                  type="text"
                  placeholder="Search a Superhero"
                  className="form-control"
                  name="searchText"
                  value={searchText}
                  onChange={onInputChange}
                  autoComplete="off"
                />
                <button className="btn btn-secondary mt-2">Search</button>
              </form>
            </div>
          </div>
          <div className="col-xs-12 col-sm col-md-7 col-lg-8">
            <div className="card-body border-start border-secondary-subtle">
              <h5 className="card-title">Search results...</h5>
              <div
                className="alert alert-secondary animate__animated animate__headShake"
                style={{ display: showSearch ? "" : "none" }}
                aria-label="alert-no-text"
              >
                <p className="p-0 m-0">
                  Please introduce the name of a superhero
                </p>
              </div>
              <div
                className="alert alert-danger animate__animated animate__headShake"
                style={{ display: showError ? "" : "none" }}
                aria-label="alert-no-hero"
              >
                <p className="p-0 m-0">
                  We could not find a hero with name <b>{q}</b>
                </p>
              </div>
              <div className="row rows-cols-xs-1 row-cols-sm-1 row-cols-md-2 row-cols-lg-3 g-3">
                {heroes.map((hero) => (
                  <HeroCard key={hero.id} {...hero} />
                ))}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
