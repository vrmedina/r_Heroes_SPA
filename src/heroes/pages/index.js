export * from "./DcPage";
export * from "./ErrorPage";
export * from "./HeroesRootPage";
export * from "./MarvelPage";