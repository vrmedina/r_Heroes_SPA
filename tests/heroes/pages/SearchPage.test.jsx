import { render, screen } from "@testing-library/react";
import { SearchPage } from "../../../src/heroes/pages/SearchPage";
import { MemoryRouter } from "react-router-dom";

describe("Pruebas en <SearchPage />", () => {
  test("debe mostrarse correctamente con valores por defecto", () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  test("debe mostrar a batman y el input con el valor del queryString", () => {
    render(
      <MemoryRouter initialEntries={["/search?q=batman"]}>
        <SearchPage />
      </MemoryRouter>,
    );

    const input = screen.getByRole("textbox");
    expect(input.value).toBe("batman");

    const img = screen.getByRole("img");
    expect(img.src).toContain("/assets/heroes/dc-batman.jpg");

    const alertNoText = screen.getByLabelText("alert-no-text");
    expect(alertNoText.style.display).toBe("none");

    screen.debug();
  });
});
