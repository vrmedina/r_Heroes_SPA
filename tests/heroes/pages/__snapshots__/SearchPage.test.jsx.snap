// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> debe mostrarse correctamente con valores por defecto 1`] = `
<div>
  <div
    class="container container-fluid"
  >
    <h1
      class="mt-3"
    >
      Superhero Search
    </h1>
    <hr />
    <div
      class="card my-3"
      style="max-width: 1000px;"
    >
      <div
        class="row g-0"
      >
        <div
          class="col-xs-12 col-sm-5 col-md-5 col-lg-4"
        >
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Search
            </h5>
            <hr />
            <form>
              <input
                autocomplete="off"
                class="form-control"
                name="searchText"
                placeholder="Search a Superhero"
                type="text"
                value=""
              />
              <button
                class="btn btn-secondary mt-2"
              >
                Search
              </button>
            </form>
          </div>
        </div>
        <div
          class="col-xs-12 col-sm col-md-7 col-lg-8"
        >
          <div
            class="card-body border-start border-secondary-subtle"
          >
            <h5
              class="card-title"
            >
              Search results...
            </h5>
            <div
              aria-label="alert-no-text"
              class="alert alert-secondary animate__animated animate__headShake"
            >
              <p
                class="p-0 m-0"
              >
                Please introduce the name of a superhero
              </p>
            </div>
            <div
              aria-label="alert-no-hero"
              class="alert alert-danger animate__animated animate__headShake"
              style="display: none;"
            >
              <p
                class="p-0 m-0"
              >
                We could not find a hero with name 
                <b />
              </p>
            </div>
            <div
              class="row rows-cols-xs-1 row-cols-sm-1 row-cols-md-2 row-cols-lg-3 g-3"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
